apps/customer/views.py

```diff
@@ -99,29 +101,37 @@ def update(self, request, *args, **kwargs):
         return update_instance(self, request, *args, **kwargs)
     
 #==========================================================================  
-
+    
 class CustomerCreateViews(APIView):
-
     def get_object(self, pk):
         try:
             return Customer.objects.get(pk=pk)
         except Customer.DoesNotExist:
             logger.warning(f"Customer with ID {pk} does not exist.")
-            return build_response(0, "Record does not exist", [], status.HTTP_404_NOT_FOUND)
+            return None
 
-    def get(self, request,  *args, **kwargs):
-        if 'pk' in kwargs:
+    def get(self, request, *args, **kwargs):
+        if "pk" in kwargs:
             result =  validate_input_pk(self,kwargs['pk'])
             return result if result else self.retrieve(self, request, *args, **kwargs)
+
         try:
-            instance = Customer.objects.all()
-        except Customer.DoesNotExist:
-            logger.error("Customer does not exist.")
-            return build_response(0, "Record does not exist", [], status.HTTP_404_NOT_FOUND)
-        else:
-            serializer = CustomerSerializer(instance, many=True)
+            summary = request.query_params.get("summary", "false").lower() == "true"
+            if summary:
+                logger.info("Retrieving customer summary")
+                customers = Customer.objects.all()
+                data = CustomerOptionSerializer.get_customer_summary(customers)
+                return Response(data, status=status.HTTP_200_OK)
```

Comment:
You should be using build_response util method here.

Reply:
for this summary we already defined in serializers.py file.